src/XenCert/StorageHandlerUtil.py

```diff
@@ -661,59 +661,90 @@ def get_lun_scsiid_devicename_mapping(targetIQN, portal):
         XenCertPrint("Failed to find any LUNs for IQN: %s and portal: %s" % (targetIQN, portal))
         return {}
 
+def parse_multipathd_config(lines):
```

Comment:
The definition of "multipath.conf" is recursively. Would this make the the parsing simpler?
For example:
```
def getNext(confStr):
    if conf_str.startswith("{"):
        word = getFirstWord(conStr)
        return word, confStr[len(word):]
    else:
        idx = getMatched(confStr)
        return confStr[0:idx+1], confStr[(idx+1):]
        

def parse_multipathd_config(confStr, outputDict):
    key, confStr = getNext(confStr)
    value, confStr = getNext(confStr)
    if isValue(value):
        outputDict[key] = value
    else:
        r = parse_multipathd_config(value,{})
        output_dict[key] = r

    return outputDict
```

Reply:
It looks shorter indeed, while I don't think it's simpler. I am glad to change to use your way if it has better maintainability. let's touch tomorrow.

Reply:
That's fine.